src/libexpr/parser-state.hh

```diff
@@ -189,6 +155,60 @@ inline void ParserState::addAttr(ExprAttrs * attrs, AttrPath && attrPath, const
     }
 }
 
+/**
+ * Precondition: attrPath is used for error messages and should already contain
+ * symbol as its last element.
+ */
+inline void ParserState::addAttr(ExprAttrs * attrs, AttrPath & attrPath, const Symbol & symbol, ExprAttrs::AttrDef && def)
```

Comment:
```suggestion
void ParserState::addAttr(ExprAttrs * attrs, AttrPath & attrPath, const Symbol & symbol, ExprAttrs::AttrDef && def)
```
Generally we can leave inlining to the compiler.

Reply:
Not here though. This is a header file.

Reply:
> Generally we can leave inlining to the compiler.

For the sake of completeness:

`inline` keyword in C++ has nothing to do with inlining per standard. Compilers do indeed tweak their inlining heuristic for inline functions, but the primary purpose is to make to possible for inline functions to circumvent ODR. Template functions and non-template members are implicitly inline, but when the member function has an out-of-class definition in a header it has to be inline.